spanner/spanner_create_instance_config.rb

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START spanner_create_instance_config]
+require "google/cloud/spanner"
+require "google/cloud/spanner/admin/instance"
+
+def spanner_create_instance_config project_id:, user_config_name:, base_config_id:
+  # project_id  = "Your Google Cloud project ID"
+  # user_config_name = "Your custom instance configuration name, The name must start with 'custom-'"
+  # base_config_id = "Base configuration ID to be used for creation, e.g projects/<project>/instanceConfigs/nam11"
+
+  instance_admin_client = Google::Cloud::Spanner::Admin::Instance::V1::InstanceAdmin::Client.new
```

Comment:
I would be better to use the admin instance exposed via the handwritten layer instead of directly using it from the generated code. 
`instance_admin_client = Google::Cloud::Spanner::Admin::Instance.instance_admin`